### PR TITLE
Cisco aci support

### PR DESCRIPTION
**Pull request by ratnakark**
Adding support for Cisco CNI Aci plugin.

### CODE REVIEW

File: playbooks/init/validate_aci_params.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+- name: Validate Aci deployment variables
+  hosts: oo_first_master
+  any_errors_fatal: true
+  tasks:
+  - name: Verify Aci deployment file variable is defined
+    fail:
+      msg: >
+         The variable aci_deployment_yaml_file is not defined.
```
Comment (rajatchopra) [inline playbooks/init/validate_aci_params.yml]: Can you give an example of the aci_deployment_yaml_file?
Perhaps through a README in the roles/aci?

Comment (ratnakark) [reply playbooks/init/validate_aci_params.yml]: I checked in a README file for ACI role

File: roles/aci/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+- name: setup firewall
+  import_tasks: firewall.yml
+
+- name: Deploy aci
+  shell: oc apply -f {{ aci_deployment_yaml_file }}
+  when: inventory_hostname == groups.oo_first_master.0
```
Comment (rajatchopra) [inline roles/aci/tasks/main.yml]: use delegate_to?
run_once: true

**Comment by CelsoSantos** [comment]
I'm having issues being able to install OKD with this integration.
Opened an issue regarding this: https://github.com/openshift/openshift-ansible/issues/9565

File: roles/aci/tasks/firewall.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- when: r_aci_node_firewall_enabled | bool and r_aci_node_use_firewalld | bool
+  block:
+  - name: Add firewalld allow rules
+    firewalld:
+      port: "{{ item.port }}"
+      permanent: true
+      immediate: true
+      state: enabled
+    when: item.cond | default(True)
+    with_items: "{{ r_aci_node_os_firewall_all }}"
```
Comment (CelsoSantos) [inline roles/aci/tasks/firewall.yml]: Shouldn't this line say `with_items: "{{ r_aci_node_os_firewall_allow }}"` instead of `firewall_all`?

Comment (CelsoSantos) [reply roles/aci/tasks/firewall.yml]: From preliminary testing, I confirm this line should indeed state `r_aci_node_os_firewall_allow`. After correcting it locally, the failed step (setting firewalld rules) did indeed pass.

**Comment by CelsoSantos** [comment]
@ratnakark, @vrutkovs

I'll be testing this tomorrow morning again, but from preliminary results of a test today, the installation fails. It seems like the install process is failing because it's trying to install ACI too soon, at a time where the cluster seems to be unavailable and not responding to `oc` requests (any of them).

I'll have logs and further info tomorrow morning to help debug the issue.

**Comment by CelsoSantos** [comment]
@vrutkovs not that I'm aware. According to [Cisco's compatibility Matrix](https://www.cisco.com/c/dam/en/us/td/docs/Website/datacenter/aci/virtualization/matrix/virtmatrix.html), supported Openshift versions are 3.6 and 3.9, don't know if that's the reason for this PR to go to `release-3.9`.

Regarding more information on errors, I have an execution log on which this install process fails, because it's trying to deploy ACI with `oc apply -f` **before** Openshift/API are available and responding to `oc` requests, thus failing and not having any retries.
The log is on [this gist](https://gist.github.com/CelsoSantos/fe0e6a6c16cf79e207381091bb2db4d1)

Should one try to add retries or wait for the API to become available?

I've forked this repo yesterday and created a branch (from `release-3.9`) to fix an error (see my comments on the code) and investigate the issue further.

**Comment by vrutkovs** [comment]
> According to Cisco's compatibility Matrix, supported Openshift versions are 3.6 and 3.9, don't know if that's the reason for this PR to go to release-3.9.

Okay, lets stick to `release-3.9` then

**Comment by vrutkovs** [comment]
Closing due to no recent activity. Please open a new PR if this change is still needed.
/close

**Pull request closed without merging**